recipes/FileUploadReactHooks/src/Main.purs

```diff
@@ -34,9 +33,12 @@ mkFileUploadComponent = do
   component "FileUploadComponent" \_ -> React.do
     fileList /\ setFileList <- useState' []
     let
-      handleChange t =
-        join <$> traverse HTMLInputElement.files (HTMLInputElement.fromEventTarget t)
-          >>= traverse_ (FileList.items >>> map File.name >>> setFileList)
+      handleChange t = case HTMLInputElement.fromEventTarget t of
+        Nothing -> pure unit
+        Just fileInput -> do
+          maybeFiles <- (HTMLInputElement.files fileInput)
+          for_ maybeFiles
+            (FileList.items >>> map File.name >>> setFileList)
```

Comment:
Does this compile? If so, it gets rid of that case statement.
```purescript
handleChange t =
  for_ (HtmlInputElement.fromEventTarget t) \fileInput -> do
    maybeFiles <- (HTMLInputElement.files fileInput)
    for_ maybeFiles (setFileList <<< map File.name <<< FileList.items)
```

Reply:
🎉 It does, and I like it 👍 678f287502f603c3e1e4318226bac9745f5c725a